src/rpc/rpc.ts

```diff
@@ -9,13 +10,28 @@ import type {
 import {
   isRpcCallMsg,
   isRpcReturnMsg,
-  isTransferable,
   toMsgPortNormalized,
 } from "./utils.ts";
 
 const DEFAULT_NAMESPACE = "rpc";
 const RPC_AGENT = Symbol("rpcAgent");
 
+// Support flags for features like streams and channels, 
+// we want to confirm if streams and channels are transferable
+// transferables aren't supported the same way across all browsers and runtimes,
+// in some runtimes it's more like partial support
+let supportFlags: Awaited<ReturnType<typeof isSupported>> = { streams: false, channel: false };
+(async () => {
+  await new Promise<void>((resolve) => {
+    isSupported()
+      .then((res) => (supportFlags = res, resolve()));
+
+    // Just in-case some runtime takes longer that we want to confirm support for streams and/or channels
+    // The `isSupported` function is supposed to be fairly fast, but just in-case it isn't we set a 3sec cutoff
+    setTimeout(() => resolve(), 3000);
+  });
+})();
```

Comment:
```typescript
(async () => {
  await new Promise<void>((resolve) => {
    isSupported()
      .then((res) => (supportFlags = res, resolve()));

    // Just in-case some runtime takes longer that we want to confirm support for streams and/or channels
    // The `isSupported` function is supposed to be fairly fast, but just in-case it isn't we set a 3sec cutoff
    setTimeout(() => resolve(), 3000);
  });
})();
```

What's the purpose of the cutoff? This IIFE seems to behave like the following simple line:

```typescript
isSupported().then((res) => supportFlags = res);
```

Reply:
Done. Sorry for the late response